### ISSUE TITLE
NewFilePlugin() permissions too broad

### ISSUE THREAD

**Issue by blockauditech** [republicprotocol/republic-go#36]
**Description**

**logger/file_plugin.go**

```
func NewFilePlugin(filePluginOptions FilePluginOptions) (Plugin, error) {
	var err error
	plugin := new(FilePlugin)
	plugin.GuardedObject = do.NewGuardedObject()
	switch filePluginOptions.Path {
	case "stdout":
		plugin.file = os.Stdout
	case "stderr":
		plugin.file = os.Stderr
	default:
		plugin.file, err = os.OpenFile(filePluginOptions.Path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666) // HERE
	}
	return plugin, err
}
```

**Expected Behavior**

Mask should be 0600 or 0640, but 0666 lets anyone on the system modify, delete or overwrite the newly created file while could lead to security risks depending on the environment.

**Steps to reproduce**

Call NewFilePlugin() with correct parameters.

**Version**

Current release

**Operating system**

Independent of OS

**Comment by loongy** [comment]
@susruth please open the PR for the fix into the develop branch. It will merge into master when develop merges later this week.

**Comment by susruth** [comment]
0640 is the permission that should have been used here. 0600 is too restrictive since other users need to process the log file.

**Comment by tok-kkk** [comment]
Fixed in [#37](https://github.com/republicprotocol/republic-go/pull/37)

**Comment by blockauditech** [comment]
Thanks for the quick fix folks!

Good luck :)
